mllib/src/main/scala/org/apache/spark/ml/feature/CountVectorizer.scala

```diff
@@ -113,7 +132,11 @@ private[feature] trait CountVectorizerParams extends Params with HasInputCol wit
   /** @group getParam */
   def getBinary: Boolean = $(binary)
 
-  setDefault(vocabSize -> (1 << 18), minDF -> 1.0, minTF -> 1.0, binary -> false)
+  setDefault(vocabSize -> (1 << 18),
+    minDF -> 1.0,
+    maxDF -> Long.MaxValue,
```

Comment:
> by @mgaido91: what about avoiding to set a default value and apply the filter only if it was set?

That'd be easy to do, but would make the code a couple lines longer.
I will do it if you think it's really better.

Reply:
Let's hear also what other people think about this. Probably in practice this is ok, but the other solution is cleaner IMHO.

Reply:
Yeah, I get it. It didn't work that way before; seems valuable only if you can avoid the whole filter stage (both values are set to filter nothing). Even there, I wonder if that makes any appreciable difference? I suppose I would have just done the straightforward thing here, and so this change looks OK to me.